kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/ClusterResolutionResult.java

```diff
@@ -6,109 +6,197 @@
 
 package io.kroxylicious.kubernetes.operator.resolver;
 
+import java.util.List;
 import java.util.Objects;
-import java.util.Set;
-import java.util.stream.Collectors;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
 import java.util.stream.Stream;
 
 import io.kroxylicious.kubernetes.api.common.LocalRef;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
 import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.operator.ResourcesUtil;
+import io.kroxylicious.kubernetes.operator.model.ResolvedCluster;
 
 /**
  * The result of a deep resolution of the dependencies of a single VirtualKafkaCluster. Describes
  * whether there were any resolution problems like dangling references, or referenced resources
  * that have a ResolvedRefs: False condition.
+ *
+ * @param cluster the cluster resource being recursively resolved
+ * @param proxyResolutionResult the resolution result for the KafkaProxy resource this cluster references
+ * @param filterResolutionResults resolution results for all filters referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
+ * @param serviceResolutionResult the resolution result for service referenced by the cluster
+ * @param ingressResolutionResults the resolution results for ingresses referenced by the cluster, in the same order they are declared in the VirtualKafkaCluster spec
  */
 public record ClusterResolutionResult(VirtualKafkaCluster cluster,
```

Comment:
What about Secrets, ConfigMaps referenced from a `VirtualKafkaCluster.spec.ingresses.tls` or `KafkaService.spec.tls`? If they're really not required then explain why in the Javadoc.

Reply:
Yeah it would make sense to resolve them here I think, as a dangling ref to a secret encountered in the aggregator should be reported back as a problem to the VKC. It just hasn't been done yet, so I'd be keen to leave it for another PR.

Reply:
+1.I started down that path when I did the the virtualcluster TLS work.  The PR got unwieldy and I stepped back to something smaller.

One unsolved issue I had was where to implement the Secret and ConfigMap content validations. By that I mean, the Kubernetes Secret has to be of type `kubernetes.io/tls or a particular data key has to end with `.pem`, `.p12` etc). Either the ClusterResolutionResult needs to expose the successfully resolved resources (so that the validation can be done in the Reconcilers), or the DependencyResolver needs to accept arbitrary validators (and have a mechanism to get back sufficient details to write a good condition message).   The former seemed more appealing.

Reply:
+1, let resolver handle lookups, referential integrity (including considering the ResolvedRefs conditions of Referents), stale status check also feels closely related.

But once we are looking into the Referents, inspecting their spec/data I think that's some other component's job.

Reply:
> But once we are looking into the Referents, inspecting their spec/data (beyond following refs) I think that's some other component's job.

Yeah I think that model works nicely. For adding checksums to Virtual clusters I was looking for something to give me the fully resolved and valid graph that I could just walk but it doesn't really exist on main and is a quite awkward to extract as things stand.

Reply:
Yep, after this branch merges it should be easy to add a `Stream<HasMetadata> ClusterResolutionResult#referents()`

Reply:
Or you can walk the different Lists of Referents if the client wants more control over order.

Reply:
> fully resolved and valid graph

I had this idea implemented earlier in this PR, but backed off it because there's already a lot of change/noise and classes to understand (ResolveCluster vs VirtualKafkaCluster vs ClusterResolutionResult). Can revisit again in a future PR.

Reply:
created #2169